test/Microsoft.Health.Fhir.Shared.Tests.Integration/Features/Operations/Reindex/ReindexJobTests.cs

```diff
@@ -241,44 +209,137 @@ public async Task GivenNoMatchingResources_WhenRunningReindexJob_ThenJobIsComple
-            _searchIndexer.Extract(Arg.Is<ResourceElement>(r => r.Id.Equals("searchIndicesPatient1"))).Returns(searchIndexValues1);
+            // Set up the values that the search index extraction should return during reindexing
+            MockSearchIndexExtraction(sampleName1, sampleName2, searchParam);
 
-            var searchIndexValues2 = new List<SearchIndexEntry>();
-            searchIndexValues2.Add(new SearchIndexEntry(searchParam, new StringSearchValue("searchIndicesPatient2")));
-            _searchIndexer.Extract(Arg.Is<ResourceElement>(r => r.Id.Equals("searchIndicesPatient2"))).Returns(searchIndexValues2);
+            CreateReindexResponse response = await SetUpForReindexing();
+
+            var cancellationTokenSource = new CancellationTokenSource();
 
+            try
+            {
+                await PerformReindexingOperation(response, OperationStatus.Completed, cancellationTokenSource);
+
+                // Rerun the same search as above
+                searchResults = await _searchService.Value.SearchAsync("Patient", queryParams, CancellationToken.None);
+
+                // This time, foo should not be dropped from the query string
+                Assert.Single(searchResults.Results);
+
+                // The foo search parameter can be used to filter for the first test patient
+                ResourceWrapper patient = searchResults.Results.FirstOrDefault().Resource;
+                Assert.Contains(sampleName1, patient.RawResource.Data);
+            }
+            finally
+            {
+                cancellationTokenSource.Cancel();
+
+                _searchParameterDefinitionManager.DeleteSearchParameter(searchParam.ToTypedElement());
+                await _testHelper.DeleteSearchParameterStatusAsync(searchParam.Url, CancellationToken.None);
+
+                await _fixture.DataStore.HardDeleteAsync(sample1.Wrapper.ToResourceKey(), CancellationToken.None);
+                await _fixture.DataStore.HardDeleteAsync(sample2.Wrapper.ToResourceKey(), CancellationToken.None);
+            }
+        }
+
+        private async Task PerformReindexingOperation(CreateReindexResponse response, OperationStatus operationStatus, CancellationTokenSource cancellationTokenSource)
+        {
+            Task reindexWorkerTask = _reindexJobWorker.ExecuteAsync(cancellationTokenSource.Token);
+            ReindexJobWrapper reindexJobWrapper = await _fhirOperationDataStore.GetReindexJobByIdAsync(response.Job.JobRecord.Id, cancellationTokenSource.Token);
+
+            int delayCount = 0;
+            while (reindexJobWrapper.JobRecord.Status != operationStatus && delayCount < 10)
+            {
+                await Task.Delay(1000);
+                delayCount++;
+                reindexJobWrapper = await _fhirOperationDataStore.GetReindexJobByIdAsync(response.Job.JobRecord.Id, cancellationTokenSource.Token);
+            }
+
+            Assert.InRange(delayCount, 0, 20);
```

Comment:
Should this be 0, 10 instead of 0, 20?

Reply:
Yes - I think it should actually be 9! As discussed, I'll set this back to 20 once we enable these tests for SQL. Fixed in 92df2a8